napari/utils/events/containers/_selection.py

```diff
@@ -25,47 +32,99 @@ class Selection(EventedSet[_T]):
     ----------
     data : iterable, optional
         Elements to initialize the set with.
-    current : Any, optional
-        The current item.
+
+    Attributes
+    ----------
+    active : Any, optional
+        The active item, if any.  An active item is the one being edited.
+    _current : Any, optional
+        The current item, if any.  This is used primarily by GUI views when
+        handling mouse/key events.
 
     Events
     ------
-    added (value: Set[_T])
-        emitted after an item or items are added to the set.
-        Will not be emitted if item was already in the set when added.
-    removed (value: Set[_T])
-        emitted after an item or items are removed from the set.
-        Will not be emitted if the item was not in the set when discarded.
-    current (value: _T, previous: _T)
+    changed (added: Set[_T], removed: Set[_T])
+        Emitted when the set changes, includes item(s) that have been added
+        and/or removed from the set.
+    current (value: _T)
+        emitted when the current item has changed.
+    active (value: _T)
         emitted when the current item has changed.
     """
 
-    def __init__(self, data: Iterable[_T] = (), current: Optional[_T] = None):
+    def __init__(self, data: Iterable[_T] = ()):
+        self._active: Optional[_T] = None
+        self.__current = None
         super().__init__(data=data)
-        self.events.add(current=None)
-        self._current = current
+        self.events.add(_current=None, active=None)
+        self.events.changed.connect(self._update_active)
+        self._update_active()
 
     def __repr__(self) -> str:
-        clsname = type(self).__name__
-        return f"{clsname}({repr(self._set)}, current={self.current})"
+        return f"{type(self).__name__}({repr(self._set)})"
+
+    def __hash__(self) -> int:
+        """Make selection hashable."""
+        return id(self)
 
     @property
-    def current(self) -> Optional[_T]:
+    def _current(self) -> Optional[_T]:
         """Get current item."""
-        return self._current
+        return self.__current
```

Comment:
Hmmm, dunders are generally discouraged except for Python-native attributes and methods. How about:

```suggestion
        return self._current_
```
?